File: 프로그래머스/0/181914. 9로 나눈 나머지/9로 나눈 나머지.cpp
#include <string>
#include <vector>

using namespace std;

int iPow_n(int n, int x = 10)
{
    int pow_res=1;
    for (int i = 0; i<n; i++) {
        pow_res *= pow_res * x;
    }
    return pow_res;
}

int solution(string number) {
    int answer = 0;
    int iNum = 0;
    char num_zero = 48; // '0' == 48
    
    for (int idx = 0; idx < number.size(); idx++) {
        // iNum += (number[idx] - num_zero) * iPow_n(idx);
        iNum += number[idx] - num_zero;
    }
    answer = iNum % 9;
    return answer;
}

File: 프로그래머스/0/181914. 9로 나눈 나머지/9로 나눈 나머지.py
def solution(number):
    answer = 0
    iNum = 0
    
    # number의 길이만큼 인덱스로 돕니다.
    for idx in range(len(number)):
        # 팩트: 파이썬은 아스키코드를 뺄 필요 없이 int()를 쓰면 문자가 바로 숫자가 됩니다.
        iNum += int(number[idx])

    answer = iNum % 9
    
    return answer
